src/shim.c

```diff
@@ -553,56 +616,46 @@ main(int argc, char **argv)
 		.io_seq_no      =  0,
 		.err_seq_no     =  0,
 		.exiting        =  false,
+		.token          =  NULL,
+		.proxy_address  =  NULL,
+		.proxy_port     =  -1,
 	};
 	int                ret;
 	struct sigaction   sa;
 	int                c;
 	bool               debug = false;
-	long long          val;
+	struct addrinfo    hints;
+	struct addrinfo   *servinfo, *addr;
+	char               port_str[20];
```

Comment:
Magic number. Why `20`? I'd use `BUFSIZE` or use a `#define` explaining why 20 bytes is sufficient.